### PR TITLE
Adding an API to validate a schema

### PR DESCRIPTION
**Pull request by snleee**
This api will convert the given schema to the same format
that we see from GET /schemas/{schemaName}. Similar to #1856.

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java [hunk 1]
```diff
@@ -129,6 +129,26 @@ public SuccessResponse addSchema(FormDataMultiPart multiPart) {
     return addOrUpdateSchema(null, multiPart);
   }
 
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/schemas/validate")
+  @ApiOperation(value = "Validate schema", notes = "This API returns the table config that matches the one you get "
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: fix the message here.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java [hunk 2]
```diff
@@ -129,6 +129,26 @@ public SuccessResponse addSchema(FormDataMultiPart multiPart) {
...
+      + "from 'GET /schema/{schemaName}'. This allows us to validate schema before apply.")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Successfully validated schema"),
+      @ApiResponse(code = 400, message = "Missing or invalid request body"),
+      @ApiResponse(code = 500, message = "Internal error")})
+  public String validateSchema (FormDataMultiPart multiPart) throws Exception {
+    try {
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: you don't need the try/catch. If it throws exception, a 5xx code will be sent out automatically

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java [hunk 3]
```diff
@@ -129,6 +129,26 @@ public SuccessResponse addSchema(FormDataMultiPart multiPart) {
...
+      Schema schema = getSchemaFromMultiPart(multiPart);
+      if (!schema.validate(LOGGER)) {
+        throw new ControllerApplicationException(LOGGER, "Invalid schema", Response.Status.BAD_REQUEST);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: I suggest changing the message to: "Invalid schema. Check controller logs"

The logs will identify which field had an error, if it is possible to do so.

Comment (snleee) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: fixed.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java (hunk 1 above)
Comment (snleee) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: fixed

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java (hunk 2 above)
Comment (snleee) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSchemaRestletResource.java]: removed

**Pull request merged**